Exception if PEPOCH is after the last TOA

I created some fake TOAs with a PEPOCH after the last TOA and PINT fails with an error in `compute_pulse_numbers`:
```
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

    def test_dmx_no_toas():
        model = get_model(
            io.StringIO(
                "\n".join(
                    [
                        par_base,
                        "DMX 15",
                        "DMX_0001 16",
                        "DMXR1_0001 58000",
                        "DMXR2_0001 59000",
                    ]
                )
            )
        )
>       toas = make_fake_toas(57000, 57900, 10, model)

tests/test_fitter_error_checking.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/pint/toa.py:579: in make_fake_toas
    ts.compute_pulse_numbers(model)
src/pint/toa.py:1299: in compute_pulse_numbers
    phases = model.phase(self, abs_phase=True) + delta_pulse_numbers
src/pint/models/timing_model.py:1000: in phase
    self.make_TZR_toa(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = AbsPhase(
    MJDParameter(   TZRMJD              UNSET,
    strParameter(   TZRSITE             UNSET,
    floatParameter( TZRFRQ              UNSET)
toas = <pint.toa.TOAs object at 0x7f5a91083d00>

    def make_TZR_toa(self, toas):
        """ Calculate the TZRMJD if one not given.
        TZRMJD = first toa after PEPOCH.
        """
        PEPOCH = self._parent.PEPOCH.quantity.mjd
        # TODO: add warning for PEPOCH far away from center of data?
>       TZRMJD = min([i for i in toas.get_mjds() if i > PEPOCH * u.d])
E       ValueError: min() arg is an empty sequence

src/pint/models/absolute_phase.py:119: ValueError
```

--- a/src/pint/models/absolute_phase.py
+++ b/src/pint/models/absolute_phase.py
@@ -112,10 +112,13 @@ class AbsPhase(PhaseComponent):
 
     def make_TZR_toa(self, toas):
         """ Calculate the TZRMJD if one not given.
+
         TZRMJD = first toa after PEPOCH.
         """
         PEPOCH = self._parent.PEPOCH.quantity.mjd
         # TODO: add warning for PEPOCH far away from center of data?
-        TZRMJD = min([i for i in toas.get_mjds() if i > PEPOCH * u.d])
+        later = [i for i in toas.get_mjds() if i > PEPOCH * u.d]
+        earlier = [i for i in toas.get_mjds() if i <= PEPOCH * u.d]
+        TZRMJD = min(later) if later else max(earlier)
         self.TZRMJD.quantity = TZRMJD.value
         self.setup()

--- a/tests/test_timing_model.py
+++ b/tests/test_timing_model.py
@@ -9,6 +9,7 @@ import astropy.units as u
 import numpy as np
 import pytest
 from pinttestdata import datadir
+from pint.toa import make_fake_toas
 
 from pint.models import (
     DEFAULT_ORDER,
@@ -294,3 +295,8 @@ def test_free_params(lines, param, exception):
     model = get_model(io.StringIO("\n".join([par_base] + lines)))
     with pytest.raises(exception):
         model.free_params = [param]
+
+
+def test_pepoch_late():
+    model = get_model(io.StringIO(par_base))
+    t = make_fake_toas(56000, 57000, 10, model=model)

